app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,33 @@
+class SessionsController < ApplicationController
+  before_action :load_user, only: :create
+
+  def new; end
+
+  def create
+    return if performed?
+
+    if @user&.authenticate params.dig(:session, :password)
```

Comment:
```suggestion
    if @user.authenticate params.dig(:session, :password)
```
Vì đã check load_user ở before action rồi, nên để vào được đây thì @user đã tồn tại. Không cần &. nữa

Reply:
Em tách ra 1 cái before action nữa để check authenticate luôn e, nếu fail thì thực thi nhánh else. Còn trong method create chỉ thực hiện code của logic sau khi success hết